static/js/src/advantage/react/components/FormikField/FormikField.tsx

```diff
@@ -0,0 +1,37 @@
+import type {
+  ComponentProps,
+  ComponentType,
+  ElementType,
+  HTMLProps,
+} from "react";
+import React, { useRef } from "react";
+
+import { Input } from "@canonical/react-components";
+import { nanoid } from "@reduxjs/toolkit";
+import { useField } from "formik";
+
+export type Props<C extends ElementType | ComponentType = typeof Input> = {
+  component?: C;
+  name: string;
+  value?: HTMLProps<HTMLElement>["value"];
+} & ComponentProps<C>;
+
+const FormikField = <C extends ElementType | ComponentType = typeof Input>({
+  component: Component = Input,
+  name,
+  value,
+  ...props
+}: Props<C>): JSX.Element => {
+  const id = useRef(nanoid());
```

Comment:
What is nanoid?

Reply:
It generates unique ids, here it generates the field ids for the label and the field so we don't need to manually provide them.

I also meant to mention that I copied this file over from maas-ui.